### DISCUSSION TITLE
How can I merge two yaml files using yq library ?

### DISCUSSION THREAD

**Discussion by GeorgeDuckman** [mikefarah/yq#1966] [Q&A] [answered]
Hello,

I am attempting to use yq as a library in my Golang application. I want to merge two yaml contents that are contained in two different string variables.

I am aware that the yq command can be used to merge two YAML files :

```shell
yq -n 'load("file1.yaml") * load("file2.yaml")'
```

To use yqlib, I am using the StringEvaluator which only takes one input. I am concatenating my two variables into one as follows:

```yaml
---
a: Hello
---
a: Goodbye
```

And I want the result to be :

```yaml
---
a: Goodbye
```

Is there a way to do this ? I would prefer not to create temporary files if possible :)
I've also tried merging by select but it returns nothing.

Here is my minimal code sample if you want to try yourself :

```go
package main

import (
	"fmt"
	"log"

	"github.com/mikefarah/yq/v4/pkg/yqlib"
)

func main() {
	yamlA := "---\n" +
		"a: Hello"

	yamlB := "---\n" +
		"a: Goodbye"

	yaml := yamlA + "\n" + yamlB

	se := yqlib.NewStringEvaluator()
	expression := "select(document_index == 0) * select(document_index == 1)"
	strung, err := se.Evaluate(expression, yaml, yqlib.NewYamlEncoder(2, true, yqlib.YamlPreferences{}), yqlib.NewGoccyYAMLDecoder())
	if err != nil {
		log.Fatal(err)
	}
	fmt.Println(strung)
}
```

Thank you in advance !

**Comment by mikefarah** [comment, marked as answer]
I've added `EvaluateAll` to the string evaluator in v4.43.1 - give that a go instead :)
